app/src/components/map/modal/itinerary-poi-review-modal.tsx

```diff
@@ -37,25 +39,92 @@ export function POIReviewDialog({
   options: ExtractOptions<"itinerary-poi-review">;
   close: () => void;
 }) {
+  const [isUpdate, setIsUpdate] = useState(false);
+  const [uploadedImages, setUploadedImages] = useState<string[]>([]);
+  
   const createReviewMutation = trpc.review.createReview.useMutation();
+  const updateReviewMutation = trpc.review.updateReview.useMutation();
```

Comment:
Split update and create into it's own dialog component. 

Youll make your life easier cus less states to worry about.